app/src/main/java/com/jerboa/ui/components/login/LoginViewModel.kt

```diff
@@ -97,22 +97,23 @@ class LoginViewModel : ViewModel() {
                     val siteVersion = siteRes.data.version
                     if (compareVersions(siteVersion, MINIMUM_API_VERSION) < 0) {
                         val message = ctx.resources.getString(
-                            R.string.dialogs_server_version_outdated_short,
+                            R.string.dialogs_server_version_outdated_warning_short,
                             siteVersion,
                         )
                         Toast.makeText(ctx, message, Toast.LENGTH_SHORT).show()
                     }
 
                     try {
                         val luv = siteRes.data.my_user!!.local_user_view
+
                         val account = Account(
                             id = luv.person.id,
                             name = luv.person.name,
                             current = true,
                             instance = instance,
                             jwt = jwt,
-                            defaultListingType = luv.local_user.default_listing_type.ordinal,
-                            defaultSortType = luv.local_user.default_sort_type.ordinal,
+                            defaultListingType = if (luv.local_user.default_listing_type != null) luv.local_user.default_listing_type.ordinal else 0,
```

Comment:
These are exactly the hacks I don't want riddling the codebase. They're just going to keep piling on top of each other until it becomes inscrutable.

Reply:
I totally understand what you mean.

I'm a shitty coder when it comes to best practices. I'm trying to come up with a better way to do it, would a try/catch statement be better in all the spots? I'll make a commit to switch it around.

Reply:
The "good" solution would be abstracting away the API entirely, having a version-agnostic frontend that can utilize a per-version backend. Jerboa then would query the frontend asking what features are supported on an instance, and adjust its behavior accordingly. That's quite a bit of engineering though since we're currently directly using the types autogenerated from the Lemmy API directly throughout the app with no abstraction.